packages/shared/src/components/cards/common/CardCover.tsx

```diff
@@ -0,0 +1,106 @@
+import React, { ReactElement, useState } from 'react';
+import classNames from 'classnames';
+import { ImageProps, ImageType } from '../../image/Image';
+import VideoImage, { VideoImageProps } from '../../image/VideoImage';
+import ConditionalWrapper from '../../ConditionalWrapper';
+import { CardImage } from '../Card';
+import { CardCoverShare } from './CardCoverShare';
+import { CommonCardCoverProps } from '../common';
+import {
+  upvoteMutationKey,
+  useMutationSubscription,
+  UseVotePostMutationProps,
+} from '../../../hooks';
+import { useActiveFeedNameContext } from '../../../contexts';
+import { useFeatureIsOn } from '../../GrowthBookProvider';
+import { feature } from '../../../lib/featureManagement';
+
+interface CardCoverProps extends CommonCardCoverProps {
+  imageProps: ImageProps;
+  videoProps?: Omit<VideoImageProps, 'imageProps'>;
+  isVideoType?: boolean;
+}
+
+export function CardCover({
+  imageProps,
+  videoProps,
+  isVideoType,
+  onShare,
+  post,
+}: CardCoverProps): ReactElement {
+  const { feedName } = useActiveFeedNameContext();
+  const [justUpvoted, setJustUpvoted] = useState(false);
+  const [hasInteracted, setHasInteracted] = useState(false);
+  const shareLoopsEnabled = useFeatureIsOn(feature.shareLoops);
+
+  useMutationSubscription({
+    matcher: ({ status, mutation }) => {
+      const key = [...upvoteMutationKey, { feedName }];
+
+      return (
+        status === 'success' &&
+        mutation?.options?.mutationKey?.toString() === key.toString()
+      );
+    },
+    callback: ({ variables }) => {
+      const vars = variables as UseVotePostMutationProps;
+
+      if (vars.id !== post.id || !shareLoopsEnabled) {
+        return;
+      }
+
+      setJustUpvoted(!!vars.vote);
+    },
+  });
+
+  if (!shareLoopsEnabled) {
+    return null;
+  }
+
```

Comment:
Are we sure we want to do this here? We're now using card covers also for video images, in this case the video image wouldn't show no?

Reply:
Yes, that's the reason we made this component, so either video or image, they will utilize the same process and conditions.

Reply:
Yup, but in this case they are set dependant on `shareLoopsEnabled`, this should only be a condition for the share overlay not normal video or image.

Reply:
Ohhh, right right. Wrong placement. Good catch right there 😂